runelite-client/src/main/java/net/runelite/client/plugins/inventorytags/InventoryTagsPlugin.java

```diff
@@ -212,23 +231,49 @@ public void onMenuOpened(final MenuOpened event)
 				return;
 			}
 
-			MenuEntry[] menuList = new MenuEntry[GROUPS.size()];
+			final List<MenuEntry> menuList = new ArrayList<>();
 			int num = 0;
 
 			for (final String groupName : GROUPS)
 			{
 				final String group = getTag(itemId);
 				final MenuEntry newMenu = new MenuEntry();
 				final Color color = getGroupNameColor(groupName);
-				newMenu.setOption(groupName.equals(group) ? MENU_REMOVE : MENU_SET);
-				newMenu.setTarget(ColorUtil.prependColorTag(groupName, MoreObjects.firstNonNull(color, Color.WHITE)));
+				if (groupName.equals(group))
```

Comment:
You should probably add the clear to the inventory button and not to each item in inventory

Reply:
When I tried adding the clear button to the inventory button, the clear button was always above the save button because it displays the options in sorted order. I asked the devchat on discord and it was suggested that I add the clear tags option when shift clicking on a tagged item. I'm going to work on that if theres no better way to get it on the inventory button.

Reply:
Imo it is fine when it is sorted. Adding clear tags option on each tagged item is worse.

Reply:
Alright, I'll get to work on getting that working. I'm just concerned there might be some people mis-clicking clear instead of save.

Reply:
You could make it so you have to confirm the clear by pressing clear again with a prompt message in the chatbox.

Just a suggestion, not sure how people would feel about that.

Reply:
So, after doing additional testing it appears that the menu options are not displayed in sorted order but instead they tend to switch places randomly. This will only occur when the client is restarted though.  They will remain in the same place while the client remains open but once its closed and reopened there is a chance that the positions will change. Not sure where to go from here.